### ISSUE TITLE
Project specific Preview-templates

### ISSUE THREAD

**Issue by Ronsku** [cocos/cocos-engine#3379]
Hi,

Something that would really help a lot when working with different projects using Cocos Creator would be to be able to have project specific "preview-templates" folders in the root of the projects, the same way as "build-templates" in project roots override the "resources\static\build-templates" default build templates.

Preview-templates folder is also in Cocos Creator application under "resources\static\preview-templates" the exact same way as "build-templates".

This sounds like a quite small change in the software (?) but would be a HUGE benefit in many cases where you have some project specific requirements for your preview templates. Now these changes can be easily applied only for builds.

I'm sure that the usability of Cocos Creator would go up by miles for many of us struggling to implement custom solutions for our preview templates.

Thank you a lot in advance and I'm always ready to answer any questions you might have.

**Comment by sommerper** [comment]
+1

**Comment by ColinCollins** [comment]
Thx for your suggestion. but I don't know where I can use this and it works.
Would you mind provide the environment for me to understand this feature?
@Ronsku

**Comment by Ronsku** [comment]
Hi @ColinCollins,

Sorry for the super late response. I did not see your message before this and was about to bump this post.

The problem is that you are not able to have custom "preview-templates" folders / game project, but you can have "build-templates" for a specific project. My request would be to support **Project Specific "preview-template" folders**

I try to explain by using images:
In your **Cocos Creator Application folder** you find both "preview-template" and "build-template" folders, in my case: _C:\CocosCreator\resources\static:
![image](https://user-images.githubusercontent.com/7983358/54603720-ba4c2380-4a4d-11e9-8202-79cddb759f84.png)

In my **Game Project**:
You can have "build-template" folder to **Overwrite** the default "build-template" folder from Cocos Creator Application folder, but you cannot overwrite the "preview-template" folder by having a custom project specific "preview-template":
![image](https://user-images.githubusercontent.com/7983358/54603886-2464c880-4a4e-11e9-8d6d-5bce05a7e4d4.png)

There are many cases depending on your project where you would like or need to adjust your development environment, preview tool and game wrapper depending on what project you are working on. Currently I have to replace the Cocos Creator Application "preview-template" folder each time i swap projects.

**TL/DR: Would it be possible to have project specific "preview-template" folders?**

**Comment by balazsnemeth** [comment]
👍

**Comment by Ronsku** [comment]
@ColinCollins Did you understand what we are trying to achieve with the project specific preview-template folder?

**Comment by ColinCollins** [comment]
Sorry for feedback so late @Ronsku , I just finished my vacation. We have discussed this kind of issue before, but we think that is not a necessary requested, so perhaps we won't add preview-template in a long time.
@jareguo  any other suggestions?

**Comment by jareguo** [comment]
I think we should add it in 2.2

**Comment by Ronsku** [comment]
@jareguo @ColinCollins I would really appreciate this. I think this is something that many doesn't realize how strong it would be. I'm also assuming that it would not be a super complicated thing to do (I might be wrong).

If we would get it in 2.2 I would be supper happy!

**Comment by mookhf** [comment]
+1 

It is important to have the same result when preview and release for developers.

**Comment by Ronsku** [comment]
Why was this closed @jareguo @knoxHuang ? :(

**Comment by knoxHuang** [comment]
This feature has been implemented and will be in version 2.2, so turn it off

**Comment by Ronsku** [comment]
> This feature has been implemented and will be in version 2.2, so turn it off

Awesome, Thank you! 💯 🎉

**Comment by mrspd** [comment]
What about this feature now?

**Comment by jareguo** [comment]
This feature landed in v2.2, see https://docs.cocos.com/creator/manual/en/advanced-topics/custom-preview-template.html
